### PR TITLE
[risk=low][RW-13148] Enable publish/unpublish of featured workspaces when not owner

### PR DESCRIPTION
**Pull request by jmthibault79**
Publish/unpublish did not work when the workspace was not owned by the admin user.  We require Owner to publish old-style featured workspaces, but we do not want to require this in the new process.

Instead of calling "update Terra ACL" as the admin user, call it as the RWB system itself, which will always have the appropriate access to do this.

Tested locally by publishing/unpublishing workspaces which are owned by my user and not owned by my user.

---
**PR checklist**

- [x] I have included an issue ID or "no ticket" in the PR title as outlined in [CONTRIBUTING.md](https://github.com/all-of-us/workbench/blob/main/.github/CONTRIBUTING.md).
- [x] I have included a risk tag of the form `[risk=no|low|moderate|severe]` in the PR title as outlined in [CONTRIBUTING.md](https://github.com/all-of-us/workbench/blob/main/.github/CONTRIBUTING.md).
- [x] I have manually tested this change and my testing process is described above.
- [x] This change includes appropriate automated tests, and I have documented any behavior that cannot be tested with code.
- [ ] I have added explanatory comments where the logic is not obvious.
- One or more of the following is true:
  - [x] This change is intended to complete a JIRA story, so I have checked that all AC are met for that story.
  - [x] This change fixes a bug, so I have ensured the steps to reproduce are in the Jira ticket or provided above.
  - [ ] This change impacts deployment safety (e.g. removing/altering APIs which are in use), so I have documented the impacts in the description.
  - [ ] This change includes a new feature flag, so I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later.
  - [ ] This change modifies the UI, so I have taken screenshots or recordings of the new behavior and notified the PO and UX designer in [Slack](https://pmi-engteam.slack.com/archives/C02MWP2RN5P).
  - [ ] This change modifies API behavior, so I have run the relevant E2E tests locally because API changes are not covered by our PR checks.
  - [ ] None of the above apply to this change.

### CODE REVIEW

File: api/src/main/java/org/pmiops/workbench/firecloud/FireCloudServiceImpl.java [hunk 1]
```diff
@@ -387,11 +395,11 @@ public RawlsWorkspaceACLUpdateResponseList updateWorkspaceACL(
   }
 
   @Override
-  public RawlsWorkspaceACL getWorkspaceAclAsService(
```
Comment (jmthibault79) [inline api/src/main/java/org/pmiops/workbench/firecloud/FireCloudServiceImpl.java:390]: moved up in file, to match `FireCloudService`

File: api/src/main/java/org/pmiops/workbench/mail/MailServiceImpl.java [hunk 2]
```diff
@@ -406,9 +407,6 @@ private void sendPublishUnpublishWorkspaceByAdminEmail(
       throws MessagingException {
     String actionType = publish ? "published" : "unpublished";
 
-    final String ownersForLogging =
```
Comment (jmthibault79) [inline api/src/main/java/org/pmiops/workbench/mail/MailServiceImpl.java:409]: made this a function, for reuse

File: api/src/main/java/org/pmiops/workbench/workspaceadmin/WorkspaceAdminServiceImpl.java [hunk 3]
```diff
@@ -554,7 +554,8 @@ private void updateACLForFeaturedWorkspace(
     var aclUpdate =
         FirecloudTransforms.buildAclUpdate(
             workspaceService.getPublishedWorkspacesGroupEmail(), accessLevel);
-    fireCloudService.updateWorkspaceACL(workspaceNamespace, firecloudName, List.of(aclUpdate));
+    fireCloudService.updateWorkspaceACLAsService(
```
Comment (jmthibault79) [inline api/src/main/java/org/pmiops/workbench/workspaceadmin/WorkspaceAdminServiceImpl.java:557]: here is the fix: make the ACL update as the Service, not as the (admin) user.

File: api/src/test/java/org/pmiops/workbench/workspaceadmin/WorkspaceAdminServiceTest.java [hunk 4]
```diff
@@ -165,7 +168,6 @@ public class WorkspaceAdminServiceTest {
     UserMapper.class,
     UserService.class,
     WorkspaceAuthService.class,
-    WorkspaceService.class
```
Comment (jmthibault79) [inline api/src/test/java/org/pmiops/workbench/workspaceadmin/WorkspaceAdminServiceTest.java:168]: this is now the individually accessible `@MockBean private WorkspaceService mockWorkspaceService;`

**Review by NehaBroad — APPROVED** [review]

**Pull request merged**